### PR TITLE
Update NixOS Python guidance

### PR DESCRIPTION
**Pull request by tfoote**
Following clarifying comments in https://github.com/ros/rosdistro/pull/38539#issuecomment-1718489729

### CODE REVIEW

**Comment by tfoote** [comment]
@lopsided98 @hacker1024 @kjeremy for awareness/review

**Review by clalancette — APPROVED** [review]
Seems reasonable to me, but I'll wait for further confirmation before merging.

**Comment by kjeremy** [comment]
Looks good to me.

File: CONTRIBUTING.md
Comment (hacker1024) [inline CONTRIBUTING.md:203]:
```suggestion
* Following the [NixOS Python Guide](https://nixos.org/manual/nixpkgs/stable/#python), use `pythonPackages` for Python 2 and `python3Packages` for Python 3 keys.
```

Erroneous capitalization?

**Pull request merged**
